tests/acceptance/stepDefinitions/filesContext.js

```diff
@@ -159,6 +160,27 @@ When('the user uploads file {string} using the webUI', function (element) {
   return client.page.filesPage().uploadFile(element)
 })
 
+When('the user uploads a folder containing following files using the webUI:', async function (files) {
+  files = files.raw().map(item => item[0])
+
+  if ((new Set(files)).size !== files.length) {
+    throw new Error(
+      `Allowing upload of multiple files in the same folder would complicate
+      other step-definitions. Please remove duplicated files and retry.`
+    )
+  }
+
+  if (files.length === 1) {
+    throw new Error('Please try uploading more than one file. Uploading only one file is not supported.')
```

Comment:
If we only upload a single file, there will only be a single folder inside the session folder, which Owncloud seems to upload the child folder instead.